src/ts_catalog/continuous_aggs_watermark.c

```diff
@@ -392,13 +407,22 @@ ts_cagg_watermark_update(Hypertable *mat_ht, int64 watermark, bool watermark_isn
 {
 	ContinuousAgg *cagg = ts_continuous_agg_find_by_mat_hypertable_id(mat_ht->fd.id);
 
+	/* If we have a real-time CAgg, it uses a watermark function. So, we have to invalidate the rel
+	 * cache to force a replanning of prepared statements. See cagg_watermark_update_internal for
+	 * more information. */
+	bool invalidate_rel_cache = !cagg->data.materialized_only;
```

Comment:
This should be after the NULL check below or am I missing something?

Reply:
Sure, it should be after the check. Moved. Thanks.